aea/cli/ipfs_hash.py

```diff
@@ -0,0 +1,491 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2021-2022 Valory AG
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""
+This script generates the IPFS hashes for all packages.
+
+This script requires that you have IPFS installed:
+- https://docs.ipfs.io/guides/guides/install/
+"""
+import collections
+import csv
+import operator
+import os
+import re
+import sys
+import traceback
+from pathlib import Path
+from typing import Collection, Dict, List, Optional, Tuple, Type, cast
+
+import click
+
+from aea.configurations.base import (
+    AgentConfig,
+    ConnectionConfig,
+    ContractConfig,
+    PackageConfiguration,
+    PackageType,
+    ProtocolConfig,
+    SkillConfig,
+    _compute_fingerprint,
+)
+from aea.configurations.loader import ConfigLoaders
+from aea.helpers.ipfs.base import IPFSHashOnly
+from aea.helpers.yaml_utils import yaml_dump, yaml_dump_all
+
+
+HASHES_FILE = "hashes.csv"
+HASH_TOOL = IPFSHashOnly()
+
+type_to_class_config = {
+    PackageType.AGENT: AgentConfig,
+    PackageType.PROTOCOL: ProtocolConfig,
+    PackageType.CONNECTION: ConnectionConfig,
+    PackageType.SKILL: SkillConfig,
+    PackageType.CONTRACT: ContractConfig,
+}  # type: Dict[PackageType, Type[PackageConfiguration]]
+
+
+def package_type_and_path(package_path: Path) -> Tuple[PackageType, Path]:
+    """Extract the package type from the path."""
+    item_type_plural = package_path.parent.name
+    item_type_singular = item_type_plural[:-1]
+    return PackageType(item_type_singular), package_path
+
+
+def _get_core_packages(root_dir: Path) -> List[Tuple[PackageType, Path]]:
+    return list(
+        map(
+            package_type_and_path,
+            [
+                root_dir / "aea" / "protocols" / "scaffold",
+                root_dir / "aea" / "connections" / "scaffold",
+                root_dir / "aea" / "contracts" / "scaffold",
+                root_dir / "aea" / "skills" / "scaffold",
+            ],
+        )
+    )
+
+
+def _get_test_packages(test_data: Path) -> List[Tuple[PackageType, Path]]:
```

Comment:
This, as mentioned before, is not allowed to be here. The framework should know nothing about the structure of the test folder.